test/cli/expected-got/test.out

```diff
@@ -115,6 +115,11 @@ test/cli/expected-got/expected-got.rb:57: Used `&.` operator on `Integer(1)`, wh
 test/cli/expected-got/expected-got.rb:59: Method `top_level_does_not_exist` does not exist on `T.class_of(<root>)` https://srb.help/7003
     59 |top_level_does_not_exist
         ^^^^^^^^^^^^^^^^^^^^^^^^
+  Got `T.class_of(<root>)` originating from:
+    test/cli/expected-got/expected-got.rb:36:
+    36 |takes_int_block do
+    37 |  ''
+    38 |end
```

Comment:
This one is slightly problematic, because if the block spans a lot of lines, it will bloat the error message.

In other changes, I've gone to lengths to avoid printing locs of entire Ruby blocks in error output. It matters a little bit less for the ErrorLine locs (it matters a ton for top-level errors, because top-level errors have their entire loc underlined in a red squiggle).

I don't know whether we can look into this and see why it was not getting an origin before and why it is now. I wonder if we can uncover that not showing this origin was somehow intentional before?

Reply:
The same claim technically applies to the `sig {void}` error locs in the next test.